src/App.tsx

```diff
@@ -1,24 +1,159 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Errors } from './components/Errors';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 6713;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [errorMessage, setErrorMessage]
+    = useState<ErrorMessage>(ErrorMessage.NONE);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number>(0);
+
+  const activeTodosLength = todos.filter((todo) => !todo.completed).length;
+  const completedTodosLength = todos.length - activeTodosLength;
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.ACTIVE:
+          return !todo.completed;
+        case FilterType.COMPLETE:
+          return todo.completed;
+        default:
+          return true;
+      }
+    })
+    );
+  }, [filterType, todos]);
```

Comment:
No need to filter todos in case is FilterType.ALL.
Also useMemo looks like useless here

Reply:
fixed